### DISCUSSION TITLE
`<A>` component to allow overriding `link={true/false}`

### DISCUSSION THREAD

**Discussion by LexSwed** [solidjs/solid-router#321] [Ideas]
I need one specific link to make a full page reload (from landing page to app page – for the request to go through middleware).

**When using `explicitLinks={false}` (default `<Router>` behaviour):**
- If I just use `<a>` client routing is used and adding `link={false}` doesn't go through SSR due to [this](https://github.com/solidjs/solid-router/blob/4ae0934b9e5825e7762456beb083162925b850ff/src/data/events.ts#L34C26-L34C26);
- If I use `<A>`, I can't specify `<A link={false}>` to prevent client routing, as it's [hardcoded](https://github.com/solidjs/solid-router/blob/4ae0934b9e5825e7762456beb083162925b850ff/src/components.tsx#L69);

**When using `explicitLinks={true}`:**
- In every place of the app I now need to use `<A>` component. Use `<a>` for my use case. This means I need my stylised link component to check which anchor it needs to render. 
This could be simplified if `link={false}` could be provided to the `<A>` component directly instead.

> Edit: I actually have to avoid adding `link=` attribute altogether, as `<a link="false">` is still intercepted as a Router link, due to `getAttribute('link')` check. Again, having customisable `true`/`false` would help here.

<details>
<summary>Maybe I could use route actions for what I try to achieve? (Edit: see first message, no)</summary>

```tsx
// /routes/login.tsx
export const route = {
  async load() {
    // can I redirect here? can I access request event here?
    await checkAndRedirect()
  },
};

// /routes/app/layout.tsx
export const route = {
  async load() {
    // can I redirect here? can I access request event here?
    await checkAndRedirect()
  },
};
```

But this means I need to setup the route loaders for every route I want to protect (here I'm using layout, but can I use a different layout?)

</details>

I didn't manage Stackblitz to work with current `solid-start`, but in case one wants to [run locally](https://stackblitz.com/edit/solidjs-templates-h16scx?file=src%2Froot.tsx).

**Comment by LexSwed** [comment]
Just checked, this will be run on the client only on navigation (so won't make an additional request):
```
// routes/login.tsx
export const route = {
  async load() {
    // using latest vinxi alpha
    const event = getRequestEvent();
    await checkAndRedirect(event)
  },
};
```

**Comment by Brendan-csel** [comment]
Yeah the previous router version was the same behaviour as `explicitLinks={true}`.

Luckly, there are a couple more things you can do to get the router to ignore a link click...

`<a target="_self">` - the router ignores anchors with any target attribute.
`<a rel="download">` or `<a rel="external">` - the router ignores these particular attribute values

This should probably get added to the README.

**Reply by AlexErrant** [reply]
[Added!](https://github.com/solidjs/solid-router/pull/467)

**Comment by pilcrowonpaper** [comment]
Had the same issue when I needed to redirect the user to an API route. The `target` and `rel` thing should probably be mentioned in the docs
